rag-toolchain/src/toolchain_embeddings/openai_embeddings.rs

```diff
@@ -105,26 +107,25 @@ impl OpenAIClient {
     /// # Returns
     /// `Vec<(String, Vec<f32>)>` - A vector of string embedding pairs the can be stored
     fn handle_success_response(
-        input_text: Vec<String>,
+        input_text: &Chunks,
```

Comment:
We shouldn't be passing refererence, Chunk is essentially just a Rc so we should clone the struct which clones the pointer